thunder/core/transforms.py

```diff
@@ -639,7 +629,6 @@ def put_grads(a, g):
 #
 # Data movement and transformation operator grads
 #
-@torchctx
```

Comment:
The removal of `@torchctx` here is interesting.

If a practitioner sets a different language context, how would these operations run in the torch language context?

Reply:
I will go through this list and check what functions require torch language context. We should use it only when needed for method resolution that is specific to torch, like `.size`.

Should the user-supplied language context be used only for trace acquisition?

Also, NumPy language is broken (https://github.com/Lightning-AI/lightning-thunder/issues/927):
```py
In [1]: import thunder.numpy
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[1], line 1
----> 1 import thunder.numpy

File ~/dev/lightning-thunder/thunder/numpy/__init__.py:7
      3 from collections.abc import Callable
      5 import numpy as np
----> 7 from thunder.core.langctx import langctx, Languages
      8 from thunder.numpy.langctx import register_method
     10 from thunder.core.proxies import TensorProxy

ModuleNotFoundError: No module named 'thunder.core.langctx'
```

Reply:
> I will go through this list and check what functions require torch language context. We should use it only when needed for method resolution that is specific to torch, like `.size`.
> 
> Should the user-supplied language context be used only for trace acquisition?

Yes. Whenever running a transform then the transform should define its language context. If the transform doesn't need to define a language context (because it doesn't access any tensor attributes) then I suppose it can get away without doing so. Another option would be to automatically use the torch language context for any transform, unless it explicitly specifies another language context.

> 
> Also, NumPy language is broken (#927):
> 
> ```python
> In [1]: import thunder.numpy
> ---------------------------------------------------------------------------
> ModuleNotFoundError                       Traceback (most recent call last)
> Cell In[1], line 1
> ----> 1 import thunder.numpy
> 
> File ~/dev/lightning-thunder/thunder/numpy/__init__.py:7
>       3 from collections.abc import Callable
>       5 import numpy as np
> ----> 7 from thunder.core.langctx import langctx, Languages
>       8 from thunder.numpy.langctx import register_method
>      10 from thunder.core.proxies import TensorProxy
> 
> ModuleNotFoundError: No module named 'thunder.core.langctx'
> ```

It's not surprising to me that the NumPy language context is broken. We could argue that we should only support the torch language context, and other practitioner-facing language contexts shouldn't even exist so we can remove some of the language context complexity, but that could be a pain later. We'd also have to think about how the clang and prim language contexts would continue to work (or not).

Reply:
I have added wrapping all functions that are registered with `register_grad` with torch language context in https://github.com/Lightning-AI/lightning-thunder/pull/924/commits/d1f71d92077b36b128327c16543d6f30a7e11a68. If someone wants to write the derivative rule in a different context they can still do that because the inner context overwrites outer contexts.